contracts/RewardManager.sol

```diff
@@ -0,0 +1,273 @@
+    
+    /**
+     * @notice Vesting schedule data associated for a user
+     * @dev Must be called directly by the beneficiary assigned the tokens in the schedule
+     * @return totalVested Total vested amount for user
+     * @return totalDrawnAmount total token drawn by user
+     * @return amountBurnt total amount burnt while pre maturely drawing
+     * @return claimable token available to be claimed
+     * @return bonusRewards tokens a user will get if nothing has been withdrawn untill endDistribution
+     * @return stillDue tokens still due (and currently locked) from vesting schedule
+     */
+    function vestingInfo(address _user)
+    public view
+    returns (uint256 totalVested, uint256 totalDrawnAmount, uint256 amountBurnt, uint256 claimable, uint256 bonusRewards, uint256 stillDue) {
+        return (
+        vestedAmount[_user],
+        totalDrawn[_user],
+        burntAmount[_user],
+        _availableDrawDownAmount(_user),
+        bonusReward[_user],
+        _remainingBalance(_user)
+        );
+    }
+
+    function _availableDrawDownAmount(address _user) internal view returns (uint256) {
+        uint256 currentTime = _getNow();
+        if (currentTime < startDistribution) {
+            return 0;
+        } else if (currentTime >= endDistribution || totalDrawn[_user] == vestedAmount[_user]) {
+            return _remainingBalance(_user);
+        }
+        else {
+            uint256 elapsedTime = currentTime.sub(startDistribution);
+            uint256 _totalVestingTime = endDistribution.sub(startDistribution);
+            return vestedAmount[_user].mul(elapsedTime).div(_totalVestingTime).sub(totalDrawn[_user]);
+        }
+    }
+
+    function _remainingBalance(address _user) internal view returns (uint256) {
+        return vestedAmount[_user].sub(totalDrawn[_user]);
+    }
+    
+    
+    /**
+     * @notice Draws down any vested tokens due
+     * @dev Must be called directly by the beneficiary assigned the tokens in the vesting 
+     */
+    function drawDown() external nonReentrant {
+        require(_getNow() > startDistribution, "Vesting not yet started");
+        return _drawDown(msg.sender);
+    }
+    
+    /**
+     * @notice Pre maturely Draws down all vested tokens by burning the preMaturePenalty
+     * @dev Must be called directly by the beneficiary assigned the tokens in the vesting 
+     */
+    function preMatureDraw() external nonReentrant {
+            address _beneficiary = msg.sender;
+            require(_remainingBalance(_beneficiary) > 0, "Nothing left to draw");
+
+            _drawDown(_beneficiary);
+            
+            (,,,,,uint256 remainingBalance) = vestingInfo(_beneficiary);
+            if(remainingBalance > 0){
+                uint256 burnAmount = remainingBalance.mul(preMaturePenalty).div(1000);
+                uint256 effectivePercentage = 1000 - preMaturePenalty;
+                uint256 effectiveAmount = remainingBalance.mul(effectivePercentage).div(1000);
```

Comment:
Can we directly subtract here ? 

`uint effectiveAmount = remainingBal.sub(burnAmount)` 

so that no rounding errors due to  multiplication / division.